src/Uno.UI/UI/Xaml/Media/ImageBrush.Android.cs

```diff
@@ -25,12 +25,37 @@ public partial class ImageBrush
 		private bool _imageSourceChanged;
 		private Windows.Foundation.Rect _lastDrawRect;
 		private readonly SerialDisposable _refreshPaint = new SerialDisposable();
+		private readonly SerialDisposable _sourceDisposable = new SerialDisposable();
+
+		internal event Action ImageChanged;
 
 		partial void OnSourceChangedPartial(ImageSource newValue, ImageSource oldValue)
 		{
+
+
+			if (newValue is not null)
+			{
+				newValue.InvalidateSource += OnInvalidateSource;
+				_sourceDisposable.Disposable = Disposable.Create(() => newValue.InvalidateSource -= OnInvalidateSource);
+
+				void OnInvalidateSource(object sdn, EventArgs args)
+				{
+					oldValue?.Dispose();
+					_imageSourceChanged = true;
+					_onImageLoaded?.Invoke();
+					ImageChanged?.Invoke();
```

Comment:
I think that those events would be raised only once the image has been loaded in the view, not as soon as the `ImageSource` has been updated (image decoding is usually async, so there is delay between "invalidated" and "loaded" - a.k.a. "the image is now visible in the view")